src/main/kotlin/app/hashers/CodeLongevity.kt

```diff
@@ -167,6 +210,57 @@ class CodeLongevity(private val serverRepo: Repo,
     }
 
     /**
+     * Scans the repo to extract code line ages.
+     */
+    fun scan() : CodeLineAges? {
+        // Load existing age data if any.
+        val istream = try { ObjectInputStream(FileInputStream(storagePath)) }
+            catch(e : Exception ) { null }
+        var storedHead = istream?.readObject() as RevCommit?
+        if (storedHead == head) {
+            return null
+        }
+
+        val ageData = (istream?.readObject() ?: CodeLineAges()) as CodeLineAges
+
+        // Update ages.
+        getLinesObservable(storedHead).blockingSubscribe { line ->
+            Logger.debug("Scanning: ${line}")
+            if (line.to.isDeleted) {
+                var age = line.age;
+                if (ageData.lastingLines.contains(line.oldId)) {
+                    age += ageData.lastingLines.remove(line.oldId)!!.age
+                }
+                val aggrAge = ageData.aggrAges.getOrPut(line.email,
+                                                        { CodeLineAges.AggrAge() } )
```

Comment:
It's ok to use 4 space ident (and not aligning with function arguments) in case a line doesn't fit 80 column limit